PostPairsTrading.md

```diff
@@ -64,28 +68,27 @@ coint:.pykx.import[`statsmodels.tsa.stattools]`:coint
-
+As we saw in the introduction, we are working in a tick architecture environment. This means that, in addition to receiving real-time prices for our indices, this architecture provides a tool to store the closing prices of our indices in our HDB at the end of the day. As obvious, establishing a connection with HDB is required:
 ```q
-t:`sym xgroup hdb(query_prices;trange;syms)
-:{[tr;syms]select from prices where date within (.z.d-tr;.z.d),sym in syms}
+hdb:hopen port
 ```
+As can be seen, we assume that the HDB process is listening at a given `port` in the local machine. We just `hopen` a connection with it and get the `hdb` handle.
 
-Now we simply need to send this function with the necessary parameters to our HDB. To do this, we open (`hopen`) a connection to our HDB process, obtaining a handle. To communicate with the process, we pass a list to the handle with the first element being the function and the subsequent elements being the parameters, once we get our data we finally group (`xgroup`) by index. Finally, let's not forget to close (`hclose`) the connection to HDB.
-
+Then, we define the query that we want the HDB to run. In this case, we declare `rs`, which takes the (n)umber of historic days and the involved (symb)ol(s) for which we want to retrieve data, as arguments.
 ```q
-tr:4*365         / date range     
-hdb:hopen port
-t:`sym xgroup hdb(query_prices;tr;syms)
-hclose h
+rs:{[tr;syms]select from prices where date within (.z.d-tr;.z.d),sym in syms}
 ```
+The body of the function might seem pretty familiar to the SQL practicioner. In fact, we are exploiting **qSQL syntax** here, which leverages a syntax similar to SQL but optimised for kdb+. It might also interesting to say that `.z.d` represents the current date, so we are interested on the `n` days back from today.
 
- In our case we are going to use closing prices to our ADF test, so we have to index (`@`) by column **close** from each pair in our table. Additionally, we take (`#`) the last **tr** days of data for both indexes, and finally apply our **fcoint** function to each (`.'`) pair of data lists.
+Now we need to send this function along with the necessary arguments to the HDB. This approach exemplifies a good practice in kdb+: keeping computations as close to the data as possible. Instead of requesting data and then applying a computation to it, we send the computation to the HDB itself so we avoid transmitting unnecessary data over the communication.
+```q
+t:`sym xgroup hdb(query_prices;4*365;syms)
+```
+To communicate with the process, we pass a list to the `hdb` handle with the first element being the function and the subsequent elements being the arguments (last 4 years & involved indexes). Once we get our data we finally `xgroup` by index.
 

```

Comment:
I miss closing the connection with `hclose`

Reply:
I think we could get rid of it, but if you find it relevant, I can add a brief paragraph at the end.